src/portfolios/portfolios.service.ts

```diff
@@ -78,4 +80,20 @@ export class PortfoliosService {
       transactionBaseDto
     );
   }
+
+  public async updatePortfolioName(
+    portfolioParams: PortfolioDto,
+    params: ModifyPortfolioDto
+  ): ServiceReturn<NumberedPortfolio> {
+    const { did, id } = portfolioParams;
+
+    if (id.lte(0)) {
+      throw new AppValidationError('Default portfolio name cannot be modified');
+    }
+
+    const { base, args } = extractTxBase(params);
+    const portfolio = (await this.findOne(did, id)) as NumberedPortfolio;
```

Comment:
(optional)

We can use function overloading to clean up this assertion. We use this pattern in the SDK in a few places.

```ts
 public async findOne(did: string): Promise<DefaultPortfolio>;
  public async findOne(did: string, portfolioId: BigNumber): Promise<NumberedPortfolio>;
  public async findOne(
    did: string,
    portfolioId?: BigNumber
  ): Promise<DefaultPortfolio | NumberedPortfolio> {
```

Only down side is moveAssets needs to get modified like:

```
    const fromId = toPortfolioId(from);
    const fromPortfolio = fromId ? await this.findOne(owner, fromId) : await this.findOne(owner);
```

but it gives us stronger types and prevents the assertion here.